### ISSUE TITLE
Add quizzes to test comprehension

### ISSUE THREAD

**Issue by willcrichton** [google/comprehensive-rust#119]
This book could potentially benefit from using the [mdbook-quiz](https://github.com/willcrichton/mdbook-quiz/) component to provide interactive quizzes that test readers' comprehension of the material. You could consider reusing the existing quizzes developed for the Rust Book experiment: https://github.com/willcrichton/rust-book/tree/main/quizzes

**Comment by mgeisler** [comment]
Hi @willcrichton,

Thanks for bringing this up! I think quizzes could be a great way to engage people further. I'm not sure when we'll get around to it, but now it's on the radar for us.

**Comment by mgeisler** [comment]
@sakex, you brought up quizzes recently — this is the infrastructure which I think we would be using

@djmitche, do I remember correctly from a discussion that you would see quizzes as a temporary stopgap measure until we find actual exercises for people?

@randomPoison and @fw-immunant, what's your take on exercises?

**Comment by djmitche** [comment]
Basically, yes -- although at this point the new course outline has an exercise for every segment, so perhaps not required.

My issue with quizzes is, they do not provide students with practice, and I think that's what students want. Quizzes are good for checking comprehension, either for one student or (if there was some way to aggregate them, like with a [student response system](https://www.iclicker.com/) for the entire class to determine whether it's OK to proceed to the next segment.

Exercises also scale nicely to different experience levels -- someone who's truly new to Rust might struggle just to get the thing to work, while someone who's trying to fill in the gaps in their Rust experience might give themselves an added challenge such as using iterators instead of loops or avoiding clones.

**Comment by randomPoison** [comment]
Yeah, I could see quizzes being useful to a limited extent. Looking at the v2 outline, I generally like that we're spreading more, smaller exercises out throughout the course. Maybe for some of the simpler sections we can do a short quiz instead of an exercise? It'd be quicker to have students answer one or two questions on their own instead of having them do a full exercise, so could also be an option for cutting down the runtime of the course where needed.
